src/ansys/fluent/core/search.py

```diff
@@ -281,10 +280,18 @@ def _write_api_tree_file(api_tree_data: dict, api_object_names: list):
                 if sys.version_info[0] < 3
                 else wn.synsets(name, lang="eng")
             )
-            synset_names = []
+            synset_names = set()
             for api_object_name_synset in api_object_name_synsets:
-                synset_names.append(api_object_name_synset.name().split(".")[0])
-            all_api_object_name_synsets[name] = synset_names
+                synset_names.add(api_object_name_synset.name().split(".")[0])
+            if synset_names:
+                unrelated_synsets = set()
+                synset_names.discard(name)
+                for synset_name in synset_names:
+                    if synset_name not in api_object_names:
```

Comment:
Not sure if I understand this logic. If `A` and `B` are api names at 2 different paths, `S1` and `S2` are synsets of `A`. If `B == S1`, will this code remove `S1` from synsets of A?

Reply:
This is the fragment from the original .json:
```
"feature": [
    "feature",
    "feature",
    "feature",
    "feature",
    "feature_of_speech",
    "feature",
    "have",
    "sport"
],
```
I thought that the algotithm will still work if we change the above fragment to:
```
"feature": [
    "feature_of_speech",
    "have",
    "sport"
],
```
It is upto you to decide the final requirement. I now have some doubts that we may actually require `feature` among the synsets because at runtime we compute the closeness between 2 sets of synsets.
E.g.,
`A` has the synsets `[A, B, S1, S2]`
`B` has the synsets `[A, B, S3, S4]`
If we remove `A` from synsets of `A`, them at runtime we cannot match `B` with `A`. Please verify this.

Reply:
@mkundu1 

1. Actually we have done this remove unrelated synsets like `mercantile_establishment` which is also not present in our API object names list.

2. In our algorithm, we are creating a list of synsets for an API object name. Then we are iterating over that list to print the search results.

Now I realized that we should not discard API name from the synsets list due to point 2. I will update this.

Reply:
Please let me know if you have still any doubts. Thank you.

Reply:
I agree with your point that "we may actually require feature among the synsets" due to point 2.

Reply:
OK, I understand the runtime behaviour now, thanks.

> We want synsets that are also present in our API object name list.

Is this a valid requirement? E.g. `shade` is not present in api names, but it is a synonym of color. If do a semantic search with `shade`, can I expect a match with the api paths for color options (e.g. contour coloring)?

Reply:
@mkundu1 Yes, this is a valid requirement. If user gives `search_string` as `shade` then it's synsets list should contains `colour`, `colouring` then we can expect a match with the API paths for colour options.

Reply:
We need to make sure that the unit testing is comprehensive so that we don't need to debate what the code might be doing -- or not doing.

Reply:
@hpohekar There can be words which are not synsets of each other but have one or more common synsets.

However, I checked the latest generated json in your branch, it contains synsets which are non-api names. So it seems to agree with what I expect.

Reply:
@mkundu1 Yes, I have updated it to contain non-api names as well, as per our above discussion.